### ISSUE TITLE
Changed behaviour for `Router` between 0.21.6 and 1.0.0-M3

### ISSUE THREAD

**Issue by esarbe** [http4s/http4s#3586]
There's a change in behavior `Router` between 0.21.6 and 1.0.0-M3, where the prefix is not properly accounted for.

Example

```
  val getRequest = Request[IO](Method.GET, uri"/prefix/foo?param=true")

  val service = HttpRoutes.of[IO] {
    case GET -> Root :? params =>
      IO(Response(Status.Ok))
    case GET -> Root / path =>
      IO(Response(Status.Accepted))
  }

  service.orNotFound.run(getRequest)

  val result = Router(
    "/prefix/foo" -> service
  ).orNotFound.run(getRequest).unsafeRunSync()

  println(result)
```

With 0.21.6 the first case (`Status.Ok`) is selected, with 1.0.0-M3 the second case (`Status.Accepted`) is selected.

**Comment by rossabaker** [comment]
Not immediately clear why this would be, but I'm going to guess it's the URI refactoring.

**Comment by hamnis** [comment]
This was fixed by #3714, apparently forgot to close it.
